packages/teams-js/src/private/videoEx.ts

```diff
@@ -117,38 +173,63 @@ export namespace videoEx {
    * Register to process video frames
    * @beta
    *
-   * @param frameCallback - The callback to invoke when registerForVideoFrame has completed
-   * @param config - VideoFrameConfig to customize generated video frame parameters
+   * @param parameters - Callbacks and configuration to process the video frames. A host may support either {@link VideoFrameHandler} or {@link VideoBufferHandler}, but not both.
+   * To ensure the video effect works on all supported hosts, the video app must provide both {@link VideoFrameHandler} and {@link VideoBufferHandler}.
+   * The host will choose the appropriate callback based on the host's capability.
    *
    * @internal
    * Limited to Microsoft-internal use
    */
   export function registerForVideoFrame(parameters: RegisterForVideoFrameParameters): void {
-    ensureInitialized(runtime, FrameContexts.sidePanel);
-    if (!isSupported()) {
-      throw errorNotSupportedOnPlatform;
+    if (!parameters.videoFrameHandler || !parameters.videoBufferHandler) {
+      throw new Error('Both videoFrameHandler and videoBufferHandler must be provided');
     }
-    if (!parameters.videoBufferHandler) {
-      throw new Error('parameters.videoBufferHandler must be provided');
+
+    if (ensureInitialized(runtime, FrameContexts.sidePanel)) {
+      if (runtime.supports.video?.mediaStream) {
+        registerHandler(
+          'video.startVideoExtensibilityVideoStream',
+          async (mediaStreamInfo: { streamId: string; metadataInTexture?: boolean }) => {
+            const { streamId, metadataInTexture } = mediaStreamInfo;
+            const generator = metadataInTexture
+              ? await processMediaStreamWithMetadata(streamId, parameters.videoFrameHandler, notifyError)
+              : await processMediaStream(streamId, parameters.videoFrameHandler, notifyError);
+            // register the video track with processed frames back to the stream
+            !inServerSideRenderingEnvironment() &&
+              window['chrome']?.webview?.registerTextureStream(streamId, generator);
+          },
+          false,
+        );
```

Comment:
The registerHandler call above is already sending a message to Teams. Why do we need a separate one here?

Reply:
This is for backward compatible reasons, the initial implementation somehow used a separate message.